nomad/server.go

```diff
@@ -353,6 +353,112 @@ func NewServer(config *Config, consulCatalog consul.CatalogAPI, logger *log.Logg
 	return s, nil
 }
 
+// Start the RPC listeners
+func (s *Server) startRPCListener() {
+	s.rpcListenerLock.Lock()
+	defer s.rpcListenerLock.Unlock()
+	ctx, cancel := context.WithCancel(context.Background())
+	s.rpcCancel = cancel
+	go func() {
+		defer close(s.listenerCh)
+		s.listen(ctx)
+	}()
+}
+
+func (s *Server) createRPCListener() error {
+	s.rpcListenerLock.Lock()
+	defer s.rpcListenerLock.Unlock()
+
+	s.listenerCh = make(chan struct{})
```

Comment:
We have two places repeating this code. It should be consolidated: https://github.com/hashicorp/nomad/blob/dd2140c12b1a79dc739ecbf4c0e90211091bbc55/nomad/server.go#L881